lib/Transformer.js

```diff
@@ -458,17 +475,29 @@ export class Transformer {
   }
 
   _getIdForTerm({term, plural}) {
-    const id = this.termToId.get(term);
+    let id;
+    // check keywords table
+    if(this.termToId.has(term)) {
+      id = this.termToId.get(term);
+    }
+
     if(id === undefined) {
+      /*
       throw new CborldError(
         'ERR_UNDEFINED_TERM',
         'CBOR-LD compression requires all terms to be defined in a JSON-LD ' +
         'context.');
+      */
+      return term;
     }
+
     return plural ? id + 1 : id;
   }
 
   _getTermForId({id}) {
+    if(typeof id === 'string') {
+      return {term: id};
```

Comment:
Should we explicitly set plural to something here? I imagine we're just relying on it being treated as `false`, so that?

```suggestion
      // dynamically generate term info for uncompressed term; `plural`
      // as `false` will cause the associated value to pass through
      // without any valence-related modifications
      return {term: id, plural: false};
```